app/models/pro_account.rb

```diff
@@ -15,10 +15,12 @@ class ProAccount < ActiveRecord::Base
              :inverse_of => :pro_account
 
   def active?
-    stripe_customer.subscriptions.any?
+    stripe_customer.present? && stripe_customer.subscriptions.any?
   end
 
   def stripe_customer
-    Stripe::Customer.retrieve(stripe_customer_id) if stripe_customer_id
+    if stripe_customer_id
+      @stripe_customer ||= Stripe::Customer.retrieve(stripe_customer_id)
+    end
```

Comment:
+1 to memoization, but you've introduced it without tests.

Usually I'd define a non-memoized version, and then memoize the result of that:

```ruby
  def stripe_customer
    @stripe_customer ||= stripe_customer!
  end

  def stripe_customer!
    Stripe::Customer.retrieve(stripe_customer_id) if stripe_customer_id
  end
```

Its also worth explaining _why_ an error is caused in the commit message. All it needs to say is that "we're calling `#subscriptions` on `nil`"